server/src/main/java/io/unitycatalog/server/persist/MetastoreRepository.java

```diff
@@ -1,19 +1,75 @@
 package io.unitycatalog.server.persist;
 
+import io.unitycatalog.server.exception.BaseException;
+import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.model.GetMetastoreSummaryResponse;
+import io.unitycatalog.server.persist.dao.MetastoreDAO;
+import io.unitycatalog.server.persist.utils.HibernateUtils;
+import io.unitycatalog.server.utils.Constants;
+
+import java.time.Instant;
+import java.util.Date;
 import java.util.UUID;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-/** The metastore is an object/resource to allow granting catalog-wide privileges. */
 public class MetastoreRepository {
-
-  private static final UUID METASTORE_ID = UUID.fromString("ca7a1095-537c-4f9c-a136-5ca1ab1ec0de");
-
   private static final MetastoreRepository INSTANCE = new MetastoreRepository();
+  private static final Logger LOGGER = LoggerFactory.getLogger(MetastoreRepository.class);
+  private static final SessionFactory SESSION_FACTORY = HibernateUtils.getSessionFactory();
+
+  private MetastoreRepository() {}
 
   public static MetastoreRepository getInstance() {
     return INSTANCE;
   }
 
+  public GetMetastoreSummaryResponse getMetastoreSummary() {
+    try (Session session = SESSION_FACTORY.openSession()) {
+      session.setDefaultReadOnly(true);
+      MetastoreDAO metastoreDAO = getMetastoreDAO(session);
```

Comment:
If metastoreDAO never changes after init, should we cache the response of `initMetastoreIfNeeded` and use that instead?